Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx

```diff
@@ -0,0 +1,261 @@
+    fetchData(competition_id).then((registrations) => {
+      setRegistrations(registrations)
+      setIsLoading(false)
+    })
+  }, [competition_id])
+
+  const { waiting, accepted, deleted } = useMemo(
+    () => partitionRegistrations(registrations),
+    [registrations]
+  )
+
+  return isLoading ? (
+    <div className={styles.list}>
+      <LoadingMessage />
+    </div>
+  ) : (
+    <>
+      <div className={styles.list}>
+        <h2> Incoming registrations </h2>
+        <RegistrationAdministrationTable
+          registrations={waiting}
+          add={(attendee) =>
+            setSelected({
+              waiting: [...selected.waiting, attendee],
+              accepted: selected.accepted,
+              deleted: selected.deleted,
+            })
+          }
+          remove={(attendee) =>
+            setSelected({
+              waiting: selected.waiting.filter(
+                (selectedAttendee) => selectedAttendee !== attendee
+              ),
+              accepted: selected.accepted,
+              deleted: selected.deleted,
+            })
+          }
+          competition_id={competition_id}
+        />
+        <h2> Approved registrations </h2>
+        <RegistrationAdministrationTable
+          registrations={accepted}
+          add={(attendee) =>
+            setSelected({
+              accepted: [...selected.accepted, attendee],
+              waiting: selected.waiting,
+              deleted: selected.deleted,
+            })
+          }
+          remove={(attendee) =>
+            setSelected({
+              accepted: selected.accepted.filter(
+                (selectedAttendee) => selectedAttendee !== attendee
+              ),
+              waiting: selected.waiting,
+              deleted: selected.deleted,
+            })
+          }
+          competition_id={competition_id}
+        />
+        <h2> Deleted registrations </h2>
+        <RegistrationAdministrationTable
+          registrations={deleted}
+          add={(attendee) =>
+            setSelected({
+              deleted: [...selected.deleted, attendee],
+              accepted: selected.accepted,
+              waiting: selected.deleted,
+            })
+          }
+          remove={(attendee) =>
+            setSelected({
+              deleted: selected.deleted.filter(
+                (selectedAttendee) => selectedAttendee !== attendee
+              ),
+              accepted: selected.accepted,
+              waiting: selected.deleted,
+            })
+          }
+          competition_id={competition_id}
+        />
+      </div>
+      <RegistrationActions
+        selected={selected}
+        refresh={() =>
+          fetchData(competition_id).then((registrations) =>
+            setRegistrations(registrations)
+          )
+        }
+      />
+    </>
+  )
+}
+
+function RegistrationAdministrationTable({
+  registrations,
+  add,
+  remove,
+  competition_id,
+}) {
+  const [checkedBoxes, setCheckedBoxes] = useState([])
```

Comment:
There's already state in the parent for which rows are selected right? It would be good to use that as the single source of truth, unless trying to set that up would be blocked by something?

Reply:
very true, changed